todo_app/app.py

```diff
@@ -1,17 +1,47 @@
-from flask import Flask, redirect, render_template, request
+from functools import wraps
+from flask import Flask, redirect, render_template, request, current_app
 from flask.helpers import flash
 from todo_app.data.item_service import ItemService
 from todo_app.flask_config import Config
 from todo_app.models.ApiException import ApiException
+from todo_app.models.User import AnonymousUser, User
 from todo_app.view_models.item_view_model import ItemViewModel
+from flask_login import LoginManager, login_required, login_user, current_user
+import requests
+import os
+
+
+def check_authorisation(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        login_disabled = current_app.config["LOGIN_DISABLED"]
+        if((not login_disabled) and current_user.get_role() != 'writer'):
+            flash("You are not authorised to do this action", "error")
+            return redirect('/')
+        return f(*args, **kwargs)
+    return decorated_function
+
 
 def create_app():
     app = Flask(__name__)
     app.config.from_object(Config())
 
     itemService = ItemService()
+    login_manager = LoginManager()
+    login_manager.anonymous_user = AnonymousUser
```

Comment:
I haven't quite come to my own conclusion on best practice for this, but worth considering the trade offs for this current model for anonymity; in particular that any endpoint not labelled as `login_required` could let through an anonymous user and will automatically assume the user is a writer - fine if we're careful but increases the potential that we accidentally miss some controls. Potentially adding a gate so that this anonymous user is only set in explicitly test environments (e.g. when `LOGIN_DISABLED` is turned on? If `FLASK_ENV` is prod it's not allowed?) could make this a little safer